creativecommons/legaldb
# Implement Pagination for Improved Navigation on Scholarship and Case Pages

## Fixes
- Fixes #227 by @abba360  

## Description
This pull request adds pagination functionality to the case and scholarship webpage, enhancing user navigation and improving load times.

## Technical details
- Implemented pagination using Django's built-in pagination features.
- Styled the pagination controls with HTML and CSS for a better user interface.
- Updated relevant views and templates to support pagination.

## Tests
1. Run the development server.
2. Navigate to the case webpage and scholarship webpage.
3. Confirm that each page displays a limited number of items (4 records per page), which can be adjusted to increase or decrease the number of items shown
4. Ensure that the pagination controls are mobile-responsive, allowing users to navigate easily on smaller screens with appropriately sized buttons and an adaptable layout
5. Check that pagination controls (Next, Previous, Page Numbers) function correctly and display the correct items.

## Screenshots
# case 
![case2](https://github.com/user-attachments/assets/40025332-47a8-470e-adac-93db5d82b460)
![case3](https://github.com/user-attachments/assets/6ee1f854-6884-41cc-8279-45288ac94b05)
![case4](https://github.com/user-attachments/assets/5491500b-ab89-4b75-a431-d02b71800371)

# case pagination mobile view
![case2-mobile](https://github.com/user-attachments/assets/5ac0f44e-2930-4877-b472-92ad64a906e8)
![case-mobile1](https://github.com/user-attachments/assets/10a3acfa-41ed-4b5c-8c6f-e790f59d81f2)
![case-mobile3](https://github.com/user-attachments/assets/05966b67-0cdc-40ad-a40a-8b3b631ef90a)

# scholarship 
![scholarship1](https://github.com/user-attachments/assets/ae45a7b8-f242-4ed4-b8b9-20ecb4448696)
![scholarship2](https://github.com/user-attachments/assets/bcc39d25-19a7-4995-a33a-2b115e31f466)
![scholarship33](https://github.com/user-attachments/assets/8393e102-7944-474b-a53f-71c0ef6286e3)


# scholarship pagination mobile view
![scholarship-mobile](https://github.com/user-attachments/assets/ad1396fb-8afe-460d-a7ca-0e590910fb24)
![scholarship-mobile1](https://github.com/user-attachments/assets/08f2a86d-4a36-4998-8e04-fa0d661db69c)
![scholarship-mobile2](https://github.com/user-attachments/assets/bd003b4d-115b-4cae-b069-6a34a4013ce7)


## Checklist



  index.md`).
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin




For the purposes of this DCO, "license" is equivalent to "license or public domain dedication," and "open source license" is equivalent to "open content license or public domain dedication."
<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
